src/plugins/intel_gna/src/gna_graph_compiler.cpp

```diff
@@ -2392,6 +2392,39 @@ void GNAGraphCompiler::CreateLayerPrimitive(CNNLayerPtr layer) {
     }
 }
 
+bool GNAGraphCompiler::should_use_scratch_allocation(const InferenceEngine::CNNLayerPtr& layer, bool fp32_mode) const {
+    auto is_non_functional_layer = [](CNNLayerPtr l) {
+        return LayerInfo(l).isNonFunctional();
+    };
+
+    auto next_layer = CNNNetCheckNextLayerSkipCertain(layer, 0, 0, true, is_non_functional_layer).first;
+    auto prev_layer = CNNNetPrevLayerSkipCertain(layer, 0, is_non_functional_layer, true);
+
+    // In FP32 mode, max pooling or activation function are implemented as separate operations, which cannot be
```

Comment:
in with this condition you will return true for fp32

Reply:
This is expected behavior. In FP32 mode, the DNN model is used, where activation and max pooling occur as separate layers, so a memory allocation for them is required.

Reply:
but function name suggest that it checks if some optimalization feature is required. Maybe this condition should be check on upper level and method should not be invoked for fp32 at all?

Reply:
I prefer to have information about memory allocation needs in one place (function) in the code.